### DISCUSSION TITLE
ESP32-Cx compile not possible

### DISCUSSION THREAD

**Discussion by Brightness248** [arendst/Tasmota#22313] [Q&A] [answered]
For some time now I have been unable to create binaries for the C series. Error code in platformio is as follows:
`...bin/as.exe': CreateProcess: No such file or directory`
in the meantime I have tried the following:
- emptied the packages folder according to the forum   
- reinstalled VSCode & platformio 
- I have also used the latest Tasmota version 

There was such a problem only in May. Something was changed in `platformio_tasmota32.ini` to solve the problem.

**Comment by InHi66** [comment]
try following:

execute in cmd:
mklink /D /J "insert path to Tasmota-development" c:\\.VSC-Tasmota

in platformio-override.ini set
core_dir = c:\\.VSC-Tasmota\\.platformio
workspace_dir = c:\\.VSC-Tasmota\\.pio

in VSCcode Platformio open Folder c:\\.VSC-Tasmota

this works to me

**Reply by Brightness248** [reply]
Hello @InHi66,

Thank you for your quick reply.

The response from the cmd command was "this already exists".
I have added it to `platformio-override.ini` as recommended.

During the build process, this time significantly more libraries were added before the same error message appeared again:

`.../riscv32-esp-elf/bin/as.exe': CreateProcess: No such file or directory`

**Comment by Jason2866** [comment]
See #21941

**Reply by Brightness248** [reply]
Hello @Jason2866,

Unfortunately, moving the .platformio folder as recommended by arendst was not successful.

I also don't understand why this problem only affects the c-series. Normal esp32 can be compiled normally.

**Reply by Jason2866** [reply]
Internal path length of the risc-v toolchain is longer than the xtensa toolchain.
My recommendation is for Windows Users to install WSL

**Reply by Brightness248** [reply]
I've had bad experiences with Linux in the past, which is why I'm not such a big fan of Linux. 
I hope that this problem will be fixed with a Windows update at some point, after all, it worked before. 
But as the saying goes: "Windows is a banana product, it only ripens after it reaches the customer."
Just kidding xD

**Reply by Jason2866** [reply]
As explained a Windows limit can not be solved "elsewhere", it has to be fixed in Windows. Since Microsoft did not do in the last 20 years I don't believe this will happen soon. All used tools are coming from Unix / Linux world so do not expect any special requirements Windows has can or will be taken care.
Once again Tasmota project can not fix this limitations from Windows.

**Reply by Brightness248** [reply]
About 3 months ago I was still able to compile for my c3 but never mind, I'm hoping for a miracle from windows xD.

I understand what you're telling me. Thank you for your support. You're doing something great with tasmota. As a hobby programmer, I'm always impressed by tasmota and the quick support.

**Reply by Wintespe** [reply]
Please check:
https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry

**Reply by Brightness248** [reply, marked as answer]
Hello @Wintespe 
I tried your version with the long path, but unfortunately that didn't work either. Maybe I did something wrong. 

I installed wsl as @Jason2866 suggested and it worked without any problems. I'm excited about the simple combination of vscode and wsl

**Reply by Wintespe** [reply]
What a pity, have you tried the following?

In the file platformio-override.ini set
```
core_dir = c:\.VSC-Tasmota\.platformio
;workspace_dir = c:\.VSC-Tasmota\.pio
```
_core_dir = c:\.VSC-Tasmota\.platformio_ installs - as far as I can see - the necessary packages locally in the working directory of the Tasmota version to be compiled

With my Windows installation (fresh and up to date) there are no problems with this setting . To find out if there is a path problem within the Tasmota compilation under Windows - with no patch at platformio-override.ini - I installed the tool **procmon** by Mark Russinovich. This makes it possible to analyze all process activities. I could not find a path problem. The maximum length of a path was less than 200 characters. AS.exe will also always be found.

It is to be expected that PlatformIO works under Linux (WSL) since it was designed for that.
